src/main/java/com/google/firebase/messaging/AndroidNotification.java

```diff
@@ -309,6 +397,191 @@ public Builder setImage(String imageUrl) {
+     * notifications can be bridged to other devices for remote display, such as a Wear 
+     * OS watch. This hint can be set to recommend this notification not be bridged.
+     *
+     * @param localOnly The "local only" flag
+     * @return This builder.
+     */
+    public Builder setLocalOnly(Boolean localOnly) {
+      this.localOnly = localOnly;
+      return this;
+    }
+
+    /**
+     * Sets the relative priority for this notification. Priority is an indication of how much of 
+     * the user's attention should be consumed by this notification. Low-priority notifications 
+     * may be hidden from the user in certain situations, while the user might be interrupted 
+     * for a higher-priority notification. The effect of setting the same priorities may differ 
+     * slightly on different platforms.
+     *
+     * @param priority The priority value, one of the values in {MIN, LOW, DEFAULT, HIGH, MAX}
+     * @return This builder.
+     */
+    public Builder setPriority(Priority priority) {
+      this.priority = priority;
+      return this;
+    }
+
+    /**
+     * Adds a vibration timing in milliseconds in the array to use. The first value in the 
+     * array indicates the duration to wait before turning the vibrator on. The next value 
+     * indicates the Duration to keep the vibrator on. Subsequent values alternate between 
+     * duration to turn the vibrator off and to turn the vibrator on. If vibrate_timings 
+     * is set and default_vibrate_timings is set to true, the default value is used instead
+     * of the user-specified vibrate_timings.
+     * A duration in seconds with up to nine fractional digits, terminated by 's'. Example: "3.5s".
+     *
+     * @param vibrateTimingInMillis vibration time in milliseconds.
+     * @return This builder.
+     */
+    public Builder addVibrateTimingsInMillis(@NonNull Long vibrateTimingInMillis) {
+      this.vibrateTimingsInMillis.add(vibrateTimingInMillis);
+      return this;
+    }
+
+    /**
+     * Adds a list of vibration timings in milliseconds in the array to use. The first value in the 
+     * array indicates the duration to wait before turning the vibrator on. The next value 
+     * indicates the Duration to keep the vibrator on. Subsequent values alternate between 
+     * duration to turn the vibrator off and to turn the vibrator on. If vibrate_timings 
+     * is set and default_vibrate_timings is set to true, the default value is used instead
+     * of the user-specified vibrate_timings.
+     * A duration in seconds with up to nine fractional digits, terminated by 's'. Example: "3.5s".
+     *
+     * @param vibrateTimingsInMillis List of vibration time in milliseconds
+     * @return This builder.
+     */
+    public Builder addAllVibrateTimingInMillis(@NonNull List<Long> vibrateTimingsInMillis) {
+      this.vibrateTimingsInMillis.addAll(vibrateTimingsInMillis);
```

Comment:
Sorry if I missed this elsewhere, but the document specifies the following format:
`A duration in seconds with up to nine fractional digits, terminated by 's'. Example: "3.5s".`
Where do we apply this formatting to vibrate_timings? Should we also validate the array to make sure it doesn't contain more than nine digits?

Reply:
I think you are right. Actually I don't need the Duration class. I can just follow the pattern used in AndroidConfig.ttl.

Will update this soon.